backend/src/v2/histories/controller/controller.ts

```diff
@@ -0,0 +1,50 @@
+import { contract } from '@jiphyeonjeon-42/contracts';
+import { P, match } from 'ts-pattern';
+import {
+  UnauthorizedError,
+  HandlerFor,
+  unauthorized,
+} from '../../shared';
+import { HistoriesService } from '../service';
+
+// mkGetMyHistories
+type GetMyDeps = Pick<HistoriesService, 'searchMyHistories'>;
+type MkGetMy = (services: GetMyDeps) => HandlerFor<typeof contract.histories.getMyHistories>;
+export const mkGetMyHistories: MkGetMy = ({ searchMyHistories }) =>
+  async ({
+    query: {
+      query, page, limit, type,
+    },
+  }) => {
+    contract.histories.getMyHistories.query.safeParse({
+      query, page, limit, type,
+    });
+    const result = await searchMyHistories({
+      query, page, limit, type,
+    });
+
+    return match(result)
+      .with(P.instanceOf(UnauthorizedError), () => unauthorized)
+      .otherwise(() => ({
+        status: 200,
+        body: result,
+      } as const));
+  };
+
+// mkGetAllHistories
+type GetAllDeps = Pick<HistoriesService, 'searchAllHistories'>;
+type MkGetAll = (services: GetAllDeps) => HandlerFor<typeof contract.histories.getAllHistories>;
+export const mkGetAllHistories: MkGetAll = ({ searchAllHistories }) => async ({
+  query: { query, page, limit, type },
+}) => {
+  const parsedQuery = contract.histories.getMyHistories.query.parse({
+    query, page, limit, type,
+  });
+  const result = await searchAllHistories(parsedQuery);
+  return match(result)
+    .with(P.instanceOf(UnauthorizedError), () => unauthorized)
+    .otherwise(() => ({
+      status: 200,
+      body: result,
+    } as const));
```

Comment:
반환값이 `UnauthorizedError | { items: [...], meta: { ... } }` 타입이기 때문에, `UnauthorizedError`를 예상하지 못한 ts-rest에서 오류를 내고 있습니다. 다음을 참고하시면 타입을 좁힐 때 도움이 될 것 같습니다.

https://github.com/gvergnaud/ts-pattern#withpattern-handler
https://github.com/gvergnaud/ts-pattern#exhaustive-otherwise-and-run